src/Items/ItemSnow.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "ItemHandler.h"
+
+
+
+
+
+class cItemSnowHandler :
+	public cItemHandler
+{
+	using Super = cItemHandler;
+
+public:
+
+	using Super::Super;
+
+private:
+
+	virtual bool OnPlacementCommit(cPlayer & a_Player, const cItem & a_HeldItem, const Vector3i a_PlacePosition, const eBlockFace a_ClickedBlockFace, const Vector3i a_CursorPosition) override
+	{
+		BLOCKTYPE Block;
+		NIBBLETYPE Meta;
+		a_Player.GetWorld()->GetBlockTypeMeta(a_PlacePosition, Block, Meta);
+
+		// Check if incrementing existing snow height:
+		if (Block == E_BLOCK_SNOW)
+		{
+			ASSERT(Meta < 7);  // BlockSnow.h ensures that if we replace a snow layer, it won't be at max height.
```

Comment:
I believe the logic goes that BlockSnow.h says DoesIgnoreBuildCollision false (if fully stacked) and true (otherwise), so we should never be replacing a Snow block that's fully stacked. Where replacing means the coordinates == non air block, which only happens when DoesIgnoreBuildCollision returns true. Let me verify.

Reply:
If this is true keep it